File: FamilyAppFlutter/android/app/src/test/java/com/familyapp/android/ManifestRobolectricTest.java
package com.familyapp.android;

import static com.google.common.truth.Truth.assertThat;

import java.io.File;
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import org.junit.Test;
import org.w3c.dom.Document;

public class ManifestRobolectricTest {
    @Test
    public void context_and_manifest() throws Exception {
        File manifestFile = new File("src/main/AndroidManifest.xml");
        assertThat(manifestFile.exists()).isTrue();

        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
        factory.setNamespaceAware(true);
        DocumentBuilder builder = factory.newDocumentBuilder();

        try (var stream = manifestFile.toURI().toURL().openStream()) {
            Document document = builder.parse(stream);
            String packageName = document.getDocumentElement().getAttribute("package");
            assertThat(packageName).contains("com.familyapp");
        }
    }
}

File: FamilyAppFlutter/android/app/src/test/java/com/familyapp/android/ManifestRobolectricTest.kt
package com.familyapp.android

import com.google.common.truth.Truth.assertThat
import org.junit.Test
import java.io.File
import javax.xml.parsers.DocumentBuilderFactory

class ManifestRobolectricTest {
    @Test
    fun context_and_manifest() {
        val manifestFile = File("src/main/AndroidManifest.xml")
        assertThat(manifestFile.exists()).isTrue()

        val docBuilder = DocumentBuilderFactory.newInstance().apply {
            isNamespaceAware = true
        }.newDocumentBuilder()

        val document = manifestFile.inputStream().use { stream ->
            docBuilder.parse(stream)
        }

        val packageName = document.documentElement.getAttribute("package")
        assertThat(packageName).contains("com.familyapp")
    }
}
